test/circuit_optimizer/test_collect_blocks.cpp

```diff
@@ -37,20 +53,63 @@ TEST(CollectBlocks, collectMultipleSingleQubitGates) {
   qc.x(0);
   qc.x(1);
   std::cout << qc << "\n";
-  qc::CircuitOptimizer::collectBlocks(qc, 1);
+  qc::CircuitOptimizer::collectBlocks(qc, 1, false);
   std::cout << qc << "\n";
   EXPECT_EQ(qc.size(), 2);
   EXPECT_TRUE(qc.front()->isCompoundOperation());
   EXPECT_TRUE(qc.back()->isCompoundOperation());
 }
 
+TEST(CollectBlocks, collectMultipleSingleQubitCliffordGates) {
+  QuantumComputation qc(2);
+  qc.h(0);
+  qc.h(1);
+  qc.t(0);
+  qc.t(1);
+  qc.x(0);
+  qc.x(1);
+  std::cout << qc << "\n";
+  qc::CircuitOptimizer::collectBlocks(qc, 1, true);
+  std::cout << qc << "\n";
+  EXPECT_EQ(qc.size(), 6);
+}
```

Comment:
The test name implies that this tests some kind of collection. However, the actual test is set up to not change a single thing. Please revise the test so that it is actually useful.